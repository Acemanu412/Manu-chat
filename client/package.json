{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "jest": {
    "setupFilesAfterEnv": [
      "./setupTests.js"
    ]
  },
  "scripts": {
    "build": "webpack --mode production",
    "build:dev": "webpack --mode development",
    "start": "webpack-dev-server --open --mode development",
    "start:live": "webpack-dev-server --open --mode development --liveReload",
    "test": "jest"
  },
  "keywords": [],
  "author": "Manuel N. Aceituno",
  "license": "ISC",
  "devDependencies": {
    "@babel/core": "^7.11.1",
    "@babel/preset-env": "^7.11.0",
    "@babel/preset-react": "^7.10.4",
    "babel-jest": "^26.3.0",
    "babel-loader": "^8.1.0",
    "cross-fetch": "^3.0.5",
    "css-loader": "^4.2.1",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.3",
    "html-webpack-plugin": "^4.3.0",
    "jest": "^26.4.1",
    "jest-environment-enzyme": "^7.1.2",
    "regenerator-runtime": "^0.13.7",
    "style-loader": "^1.2.1",
    "webpack": "^5.0.0-beta.22",
    "webpack-cli": "^3.3.12",
    "webpack-dev-server": "^3.11.0"
  },
  "dependencies": {
    "@apollo/client": "^3.1.3",
    "graphql": "^15.3.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "shards-react": "^1.0.3",
    "subscriptions-transport-ws": "^0.9.18"
  }
}
