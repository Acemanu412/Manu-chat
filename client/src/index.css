body {
    font-family: Arial, Helvetica, sans-serif;
  }

.container{
  min-height: 90vh;
  display:flex;
  flex-flow: column;
}

.input{
  margin-top: auto;
  margin-left: auto;
  margin-right: auto;
  padding: 0 15px 0 15px;
  width: 100%;
}

.message {
  display: flex;
  border-radius: 1em;
  margin: 1em;
}

.messageForm{
  flex-grow:10;
}

.message.mine{
  justify-content: flex-end;
}

.message.notMine{
  justify-content: flex-start;
}

.messageContent{
  max-width: 50%;
  padding: 1em;
  border-radius: 1em;
}
.messageContent.mine {
  background: #6D454C;
  color: #D7FFAB
}

.messageContent.notMine {
  background: #96ADC8;
  color: #0D1B1E;
}

.userLetters {
  height: 50;
  width: 50;
  border: solid 3px #8B635C;
  border-radius: 25px;
  text-align: center;
  font-size: 18;
  padding-top:7px;
  margin-left: 1em;
  margin-right: 1em;
}

.userForm{
  flex-grow: 1;
}