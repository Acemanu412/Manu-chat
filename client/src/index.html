<html>
    <head>
        <link rel="shortcut icon" href="#">
    </head>
    <body>
      <div id="app"></div>
    </body>
</html>