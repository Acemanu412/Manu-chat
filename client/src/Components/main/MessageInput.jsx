import React, { useState } from 'react';
import { Form, FormInput, Button } from 'shards-react';
import { useMutation } from '@apollo/client';

import { POST_MESSAGE } from '../../Mutations/messages';

const MessageInput = ({ user, setUser }) => {
    const [message, setMessage] = useState('');
    const [postmessage] = useMutation(POST_MESSAGE)

    const handleSubmit = (e) => {
        e.preventDefault();
        if(!e.target[0].value) return null;
        setMessage('');
        postmessage({ variables: {user: user, content: message }})
    };

    return (
        <div className='input'>
            <Form inline={true} onSubmit={(e) => handleSubmit(e)}>

                    <FormInput 
                    className={'messageForm'}
                    placeholder="Write your message!"
                    label="Message"
                    value={message}
                    onChange={(e) => setMessage(e.target.value)}
                    />

                    <FormInput 
                    className={'userForm'}
                    placeholder="Name"
                    label="User"
                    value={user}
                    onChange={(e) => setUser(e.target.value)}
                    />

                <Button type='submit'>Send</Button>
            </Form>
        </div>
    )
};

export default MessageInput;