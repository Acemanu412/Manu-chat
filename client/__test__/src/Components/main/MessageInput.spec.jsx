import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from '@apollo/client/testing';
import { Form, FormInput, Button } from 'shards-react';
import { act } from 'react-dom/test-utils';

import MessageInput from '../../../../src/Components/main/MessageInput';
import mocks from '../../../../__mocks__/messages-mocks';

describe('MessageInput component', () => {
    const user = 'Manu';
    const setUser = jest.fn(name => { return name});
    const changeEvent = {
        target: { value: "Test"},
    };
    const submitEvent = {
        target:[{ value: "Test" }],
    }
    const addTypename = false;
    const postMessage = jest.fn( () => ({ data: {}}));
    mocks[1].result = postMessage;
    const wrapper = mount(<MockedProvider mocks={mocks} addTypename={addTypename}>
                            <MessageInput user={user} setUser={setUser} />
                        </MockedProvider>)

    it('should render', () => {
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render 1 Form, 2 FormInputs, and 1 Button', () => {
        expect(wrapper.find(Form).length).toBe(1);
        expect(wrapper.find(FormInput).length).toBe(2);
        expect(wrapper.find(Button).length).toBe(1);
    });

    it('should render a form with an initial value of the user prop passed', () => {
        expect(wrapper.find('input[label="User"]').props().value).toBe(user);
    });

    it('should execute the setUser function when User input changes', () => {
        let userInput = wrapper.find('input[label="User"]');

        userInput.simulate('change', changeEvent);
        
        expect(setUser).toBeCalledTimes(1);
        expect(setUser).toBeCalledWith(changeEvent.target.value);
    });

    it('should empty the Value from Message Input and use the POST_MESSAGE mutation', async () => {
        let messageInput = wrapper.find('input[label="Message"]');
        messageInput.props().value = "Test";

        wrapper.find('form').simulate('submit', submitEvent);

        await act( async() => await new Promise(resolve => setTimeout(resolve, 0)));
        wrapper.update();
        
        expect(postMessage).toBeCalledTimes(1);
        expect(wrapper.find('input[label="Message"]').props().value).toBe('');
    })
});