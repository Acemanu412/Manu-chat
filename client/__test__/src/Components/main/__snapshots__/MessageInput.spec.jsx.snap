// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput component should render 1`] = `"<div class=\\"input\\"><form class=\\"form-inline\\"><input placeholder=\\"Write your message!\\" label=\\"Message\\" class=\\"messageForm form-control\\" value=\\"\\"><input placeholder=\\"Name\\" label=\\"User\\" class=\\"userForm form-control\\" value=\\"Manu\\"><button type=\\"submit\\" class=\\"btn btn-primary\\">Send</button></form></div>"`;
